relay-server/src/actors/processor.rs

```diff
@@ -1061,15 +1065,28 @@ impl EnvelopeProcessorService {
 
     /// Remove profiles from the envelope if they can not be parsed
     fn filter_profiles(&self, state: &mut ProcessEnvelopeState) {
+        let mut found_profile = false;
         state.managed_envelope.retain_items(|item| match item.ty() {
-            ItemType::Profile => match relay_profiling::parse_metadata(&item.payload()) {
-                Ok(_) => ItemAction::Keep,
-                Err(err) => ItemAction::Drop(Outcome::Invalid(DiscardReason::Profiling(
-                    relay_profiling::discard_reason(err),
-                ))),
-            },
+            ItemType::Profile => {
+                if !found_profile {
+                    // Found first profile, keep it.
+                    found_profile = true;
```

Comment:
Relay also considers towards outcomes dropped profiles that could not parsed, even if they aren't dropped by dynamic sampling. https://github.com/getsentry/relay/pull/2054 mentions only "valid" profiles should be considered:

>In other words, "processed profiles" should count all valid, non rate limited profiles seen by Relays, even if they were dropped by dynamic sampling.

Reply:
The wording in that PR description is too vague, it should say: "_accepted_ processed profiles should count all valid non rate limited profiles seen by Relays" .

A better way to think of it is two data streams splitting ways (see diagram [here](https://github.com/getsentry/relay/pull/2166)), and getting counted towards a different category regardless of whether they're accepted, rate limited, or invalid.